### PR TITLE
Fix/login response body

### PR DESCRIPTION
**Pull request by artstorm**

### CODE REVIEW

File: api/database/factories/ModelFactory.php [hunk 1]
```diff
@@ -85,24 +85,15 @@
...
     $jwtAuth = \App::make('Tymon\JWTAuth\JWTAuth');
     $token = $jwtAuth->fromUser($user);
 
-    return compact('token') + $body;
+    // Transform the user to be injected in the token for the api doc.
+    $transformerService = App::make(App\Services\TransformerService::class);
+    $transformer = new App\Http\Transformers\IlluminateModelTransformer($transformerService);
+    $user = $transformer->transform($user);
+
+    return ['token' => $token, 'user' => $user];
```
Comment (zakhenry) [inline api/database/factories/ModelFactory.php]: If this transformation is only used by the documentation, then the response structure should just be constructed in the documentation eg.

```
{!!
    [
        token: $factory->make(\App\Models\AuthToken::class)->token,
        decodedToken:  ???
    ]
!!}
```

Note that the frontend does not actually read the decoded token body at all - it decodes the token itself. We only include the decoded token body for debugging purposes. If it is easier to include the whole token data then go for that.

Comment (artstorm) [inline api/database/factories/ModelFactory.php]: @xiphiaz Yes, the transformation was only for the documentation purposes, so the output matches the real output 100%. Though if it's for debugging purposes only, I'll remove that, as then it shouldn't really matter.

File: api/app/Extensions/JWTAuth/UserClaim.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    /**
+     * Validate the user claim.
+     *
+     * @param  mixed  $value
+     * @return boolean
+     */
+    protected function validate($value)
+    {
+        return is_array($value) || is_null($value);
```
Comment (zakhenry) [inline api/app/Extensions/JWTAuth/UserClaim.php]: This should also not accept a null value. If there is no user, the token is not valid.

**Comment by artstorm** [comment]
@xiphiaz I believe I have got the different things discussed implemented now. So please take a new review of the PR.

File: api/app/Http/Transformers/AuthTokenTransformer.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    /**
+     * Collections are not used for token transformations.
+     *
+     * @param  mixed  $collection
+     * @return void
+     */
+    public function transformCollection($collection)
+    {
```
Comment (Redjik) [inline api/app/Http/Transformers/AuthTokenTransformer.php:37]: throw new NotImplementedException

File: api/app/Extensions/JWTAuth/UserClaim.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    public function setValue($value)
+    {
+        // Transform the user before encoding
+        $transformerService = App::make(TransformerService::class);
+        $transformer = new IlluminateModelTransformer($transformerService);
+        $value = $transformer->transform($value);
+
+        return parent::setValue($value);
+    }
```
Comment (zakhenry) [inline api/app/Extensions/JWTAuth/UserClaim.php:36]: This is a much cleaner setup.

**Comment by zakhenry** [comment]
@artstorm I've made a few updates as I found the documentation was not including the `token: <token>` key in the responses. I ended up re-instating the model factory as I think it makes for tidier documentation. Please take a look and let me know if you see any glaring mistakes I've made, then I'll merge this as I'm much happier with the adjustments to the approach that you've made.

**Pull request merged**